lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -124,9 +112,6 @@ def self.included(mod)
       # Create a new document with source if `document_id` doesn't exist in Elasticsearch
       mod.config :doc_as_upsert, :validate => :boolean, :default => false
 
-      # DEPRECATED This setting no longer does anything. It will be marked obsolete in a future version.
-      mod.config :max_retries, :validate => :number, :default => 3
```

Comment:
This should be marked `:obsolete => "This setting no longer does anything and has been removed. Remove it from your configuration."`

Because it's currently labeled deprecated, we should mark it obsolete before removing it completely.